magma_access_gateway_installer/__init__.py

```diff
@@ -2,10 +2,57 @@
 # Copyright 2022 Canonical Ltd.
 # See LICENSE file for licensing details.
 
-from .agw_installer_ubuntu import AGWInstallerUbuntu
+import sys
+from argparse import ArgumentParser
+from ipaddress import ip_address, ip_network
+
+import netifaces  # type: ignore[import]
+
+from .agw_network_configurator import AGWInstallerNetworkConfigurator
+from .agw_preinstall_checks import AGWInstallerPreinstallChecks
+from .agw_service_user_creator import AGWInstallerServiceUserCreator
+
+
+def _cli_arguments_parser(args):
+    cli_options = ArgumentParser()
+    cli_options.add_argument(
+        "--ip-address",
+        dest="ip_address",
+        required=False,
+        help="Statically allocated SGi interface IP address. Example: 1.1.1.1/24",
+    )
+    cli_options.add_argument(
+        "--gw-ip-address",
+        dest="gw_ip_address",
+        required=False,
+        help="Upstream router IP for SGi interface. Example: 1.1.1.200",
+    )
+    return cli_options.parse_args(args)
 
 
 def main():
-    agw_installer = AGWInstallerUbuntu()
-    agw_installer.preinstall_checks()
-    agw_installer.create_magma_service_user()
+    args = _cli_arguments_parser(sys.argv[1:])
+    if args.ip_address and not args.gw_ip_address:
+        raise ValueError("Upstream router IP for SGi interface is missing! Exiting...")
+    elif not args.ip_address and args.gw_ip_address:
+        raise ValueError("SGi interface IP address is missing! Exiting...")
+    elif args.ip_address and args.gw_ip_address:
+        try:
+            ip_network(args.ip_address, False)
+            ip_address(args.gw_ip_address)
+        except ValueError as e:
```

Comment:
Why do we want to specifically raise for this error but ignore any other error?

Reply:
What I mean is if example `ip_address` fails for whatever other error, we won't catch it and the code will continue like nothing happened. Is that intended?

Reply:
As discussed, not caught exceptions are gonna be raised anyway (just not handled), so this wouldn't mask any errors. Anyway, since I'm just re-raising the exception without any special handling, it might have been confusing indeed.
Changed to catch all Exceptions.